app/models/miq_queue_worker_base/runner.rb

```diff
@@ -64,6 +64,7 @@ def get_message_via_drb
         _log.debug("#{log_prefix} #{MiqQueue.format_short_log_msg(msg)} stale, retrying...")
         next
       rescue => err
+        msg.update_column(:state, MiqQueue::STATUS_ERROR)
```

Comment:
So @Fryguy I was debating if I should just delete the row if this fails but I really can't understand how this line could throw an error if we retrieved the `msg` on line 45.  We're bypassing all other columns and validations.  If we fail here, the only recourse is to raise.  Who knows if a delete of the row would work either.